request/request.go

```diff
@@ -128,6 +161,27 @@ func makeJSONPutRequest(ctx context.Context, url string, headers []Header, body
 	return addAcceptJSONHeaders(req, headers)
 }
 
+func makeDeleteRequest(ctx context.Context, url string, headers []Header) (*http.Request, error) {
+	req, err := http.NewRequestWithContext(ctx, http.MethodDelete, url, nil)
+	if err != nil {
+		return nil, fmt.Errorf("error creating request: %w", err)
+	}
+
+	headers = append(headers, Header{Key: "Content-Type", Value: "text/plain"})
```

Comment:
`DELETE` requests don't have a body, so it doesn't make sense to have a `Content-Type`. An `Accept` header might make sense, assuming you'll do parsing, but the `Content-Type` is definitely out of place.

Reply:
+1